compiler/rustc_hir_analysis/messages.ftl

```diff
@@ -373,19 +373,20 @@ hir_analysis_start_not_target_feature = `#[start]` function is not allowed to ha
 hir_analysis_start_not_track_caller = `#[start]` function is not allowed to be `#[track_caller]`
     .label = `#[start]` function is not allowed to be `#[track_caller]`
 
-hir_analysis_static_mut_ref = reference of mutable static is disallowed
-    .label = reference of mutable static
-    .note = mutable statics can be written to by multiple threads: aliasing violations or data races will cause undefined behavior
+hir_analysis_static_mut_ref = creating a {$shared}reference to a mutable static
+    .label = {$shared}reference to mutable static
+    .note = this shared reference has lifetime `'static`, but if the static ever gets mutated, or a mutable reference is created, then any further use of this shared reference is Undefined Behavior
+    .note_mut = this mutable reference has lifetime `'static`, but if the static gets accessed (read or written) by any other means, or any other reference is created, then any further use of this mutable reference is Undefined Behavior
     .suggestion = shared references are dangerous since if there's any kind of mutation of that static while the reference lives, that's UB; use `addr_of!` instead to create a raw pointer
     .suggestion_mut = mutable references are dangerous since if there's any other pointer or reference used for that static while the reference lives, that's UB; use `addr_of_mut!` instead to create a raw pointer
```

Comment:
```suggestion
    .suggestion = use `addr_of!` instead to create a raw pointer
    .suggestion_mut = use `addr_of_mut!` instead to create a raw pointer
```
I think I had that in my earlier suggestions as well?
The part I removed just repeats the `note`, or does it add anything?

(Same change for the lint below)